datagateway_api/src/search_api/query_filter_factory.py

```diff
@@ -103,87 +103,98 @@ def get_where_filter(where_filter_input, entity_name):
-            boolean_operator = list(where_filter_input.keys())[0]
-            conditions = list(where_filter_input.values())[0]
-            conditional_where_filters = []
-
-            for condition in conditions:
-                # Could be nested AND/OR
-                where_filter = {
-                    "filter": {"where": condition},
-                }
-                conditional_where_filters.extend(
-                    SearchAPIQueryFilterFactory.get_query_filter(
-                        where_filter, entity_name,
-                    ),
+        if where_filter_input:
+            if (
+                list(where_filter_input.keys())[0] == "and"
+                or list(where_filter_input.keys())[0] == "or"
+            ):
+                log.debug(
+                    "and/or operators found: %s", list(where_filter_input.keys())[0],
                 )
+                boolean_operator = list(where_filter_input.keys())[0]
+                conditions = list(where_filter_input.values())[0]
+                conditional_where_filters = []
+
+                for condition in conditions:
+                    # Could be nested AND/OR
+                    where_filter = {
+                        "filter": {"where": condition},
+                    }
+                    conditional_where_filters.extend(
+                        SearchAPIQueryFilterFactory.get_query_filter(
+                            where_filter, entity_name,
+                        ),
+                    )
 
-            nested = NestedWhereFilters(
-                conditional_where_filters[:-1],
-                conditional_where_filters[-1],
-                boolean_operator,
-                SearchAPIQuery(entity_name),
-            )
-            where_filters.append(nested)
-        elif list(where_filter_input.keys())[0] == "text":
-            log.debug("Text operator found within JSON where object")
-            try:
-                entity_class = getattr(search_api_models, entity_name)
-            except AttributeError as e:
-                raise SearchAPIError(
-                    f"No text operator fields have been defined for {entity_name}"
-                    f", {e.args}",
+                nested = NestedWhereFilters(
+                    conditional_where_filters[:-1],
+                    conditional_where_filters[-1],
+                    boolean_operator,
+                    SearchAPIQuery(entity_name),
                 )
+                where_filters.append(nested)
+            elif list(where_filter_input.keys())[0] == "text":
+                log.debug("Text operator found within JSON where object")
+                try:
+                    entity_class = getattr(search_api_models, entity_name)
+                except AttributeError as e:
+                    raise SearchAPIError(
+                        f"No text operator fields have been defined for {entity_name}"
+                        f", {e.args}",
+                    )
```

Comment:
Agreed, that message doesn't fit the code wrapped around it. We should probably rewrite that message because it still could happen (e.g. if you call `get_where_filter()` with `BananaEntity` as the `entity_name`). Something along the lines of the following would fit better (feel free to reword):

```python
f"No model for {entity_name} could be found, a different entity name should be used."
```

As for line 156, I guess technically an exception _could_ be raised, but all the entity classes in `models.py` have `_text_operator_fields` defined in them, even if that entity has no text operator fields (an empty list is stored in that case).  In that situation, line 160 (and the logging in that conditional) covers that so I don't think any additional code changes are needed for that section?

In summary, I think a re-wording of that error message will suffice

Reply:
Fixed in https://github.com/ral-facilities/datagateway-api/pull/326/commits/263ad34be42fe57e1e7ebf67e0a534d505b78a2a